src/functionalTest/kotlin/kotlinx/kover/test/functional/framework/starter/Example.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.test.functional.framework.starter
+
+import kotlinx.kover.test.functional.framework.checker.*
+import kotlinx.kover.test.functional.framework.common.*
+import kotlinx.kover.test.functional.framework.configurator.*
+import kotlinx.kover.test.functional.framework.runner.*
+import org.junit.jupiter.api.extension.*
+import org.junit.jupiter.params.*
+import org.junit.jupiter.params.provider.*
+import java.io.*
+import java.lang.reflect.*
+import java.nio.file.*
+import java.util.stream.*
+
+@Target(AnnotationTarget.FUNCTION)
+@Retention(AnnotationRetention.RUNTIME)
+@ArgumentsSource(ExampleTestArgumentsProvider::class)
+@ParameterizedTest(name = "{0}")
+@ExtendWith(ExampleInterceptor::class)
+internal annotation class ExamplesTest(
+    val includes: Array<String> = [],
+    val excludes: Array<String> = [],
+    val commands: Array<String> = ["build"]
+)
+
+private const val EXAMPLES_DIR = "examples"
+private const val TMP_PREFIX = "kover-example-"
+
+private class ExampleTestArgumentsProvider : ArgumentsProvider {
+
+    override fun provideArguments(context: ExtensionContext): Stream<out Arguments> {
+        val annotation = (context.element.orElse(null)?.getAnnotation(ExamplesTest::class.java)
+            ?: throw IllegalStateException("Test not marked by '${ExamplesTest::class.qualifiedName}' annotation"))
+
+        val excludes = annotation.excludes.toList()
+        val includes = annotation.includes.toList()
+        val commands = annotation.commands.toList()
+
+        val files = File(EXAMPLES_DIR).listFiles { it ->
+            val name = it.name
+            it.isDirectory && !excludes.contains(name) && (includes.isEmpty() || includes.contains(name))
+        }?.toList() ?: emptyList()
+
+        return files.stream().map { ExampleArgs(it, commands) }
+    }
+}
+
+private class ExampleArgs(private val exampleDir: File, private val commands: List<String>) : Arguments {
+    override fun get(): Array<Any> {
```

Comment:
Unfortunately, it will not be possible to transfer this above, because the `CheckerContext` should already get into the arguments of the function, for argument we need to prepare a directory.

Judging by the logs, the arguments are calculated once before the test starts.